fancyimpute/solver.py

```diff
@@ -188,3 +196,29 @@ def complete(self, X):
         X_result = self.project_result(X=X_result)
         X_result[observed_mask] = X_original[observed_mask]
         return X_result
+
+    def fit(self, X, y=None):
+        """
+        Fit the imputer on input `X`.
+
+        Note: all imputations should have a `fit_transform` method,
+        but only some (like IterativeImputer) also support inductive mode
+        using `fit` or `fit_transform` on `X_train` and then `transform`
+        on new `X_test`.
+        """
+        raise ValueError("%s.fit not implemented! This imputation algorithm likely"
```

Comment:
Perhaps add "Use %s.fit_transform instead." at the end of error message.

Reply:
I think the problem is that you NEED `fit` and then `transform` if you have training and then test data. So `fit_transform` is not an actual substitute. Maybe I can see "Only %s.fit_transform is supported at this time."?